bigbluebutton-html5/imports/ui/components/user-list/user-list-content/user-participants/user-list-item/user-icons/component.jsx

```diff
@@ -1,41 +1,50 @@
-import React from 'react';
+import React, { memo } from 'react';
 import PropTypes from 'prop-types';
-import Icon from '/imports/ui/components/icon/component';
+import SlowConnection from '/imports/ui/components/slow-connection/component';
+import Auth from '/imports/ui/services/auth';
 import { styles } from './styles';
 
+const METEOR_SETTINGS_APP = Meteor.settings.public.app;
+
+const SLOW_CONNECTIONS_TYPES = METEOR_SETTINGS_APP.effectiveConnection;
+const ENABLE_NETWORK_INFORMATION = METEOR_SETTINGS_APP.enableNetworkInformation;
+
 const propTypes = {
+  isModerator: PropTypes.bool.isRequired,
   user: PropTypes.shape({
     name: PropTypes.string.isRequired,
     isPresenter: PropTypes.bool.isRequired,
     isVoiceUser: PropTypes.bool.isRequired,
     isModerator: PropTypes.bool.isRequired,
     image: PropTypes.string,
   }).isRequired,
-  compact: PropTypes.bool.isRequired,
 };
 
 const UserIcons = (props) => {
   const {
-    user,
-    compact,
+    isModerator,
+    user: {
+      effectiveConnectionType,
+      id,
+    },
   } = props;
 
-  if (compact || user.isSharingWebcam) {
-    return null;
-  }
+  const showNetworkInformation = ENABLE_NETWORK_INFORMATION
+    && SLOW_CONNECTIONS_TYPES.includes(effectiveConnectionType)
+    && (id === Auth.userID || isModerator);
 
   return (
     <div className={styles.userIcons}>
       {
-        user.isSharingWebcam ?
+        showNetworkInformation ? (
           <span className={styles.userIconsContainer}>
-            <Icon iconName="video" />
+            <SlowConnection effectiveConnectionType={effectiveConnectionType} iconOnly />
           </span>
-          : null
+        ) : null
       }
     </div>
   );
 };
 
 UserIcons.propTypes = propTypes;
-export default UserIcons;
+export default memo(UserIcons);
```

Comment:
:clap: https://scotch.io/tutorials/react-166-reactmemo-for-functional-components-rendering-control#toc-what-is-react-memo- using react memo